Cannot start recaf 2.9.0

**Describe the bug**

Cannot start recaf because an exception was thrown.

**To Reproduce**

JDK version: 12.0.1
System: Windows 10
Steps to reproduce the behavior:
1. Open with cmd terminal for the first time
2. See the error

**Exception**
Full log:
```
[main] INFO : Missing JavaFX dependencies, attempting to patch in missing classes
[main] INFO :  - Local cache found!
[main] INFO :  - Loading dependencies...
[main] INFO :  - Done!
[main] INFO : Recaf-2.9.0
[main] TRACE: Loading configuration
[main] ERROR: An error occurred loading the plugins
java.lang.IllegalStateException: No 'parent' field in java.lang.ClassLoader
        at me.coley.recaf.util.VMUtil.setParent(VMUtil.java:151)
        at me.coley.recaf.plugin.PluginFolderSource.load(PluginFolderSource.java:58)
        at org.plugface.core.impl.DefaultPluginManager.loadPlugins(DefaultPluginManager.java:105)
        at me.coley.recaf.plugin.PluginsManager.load(PluginsManager.java:56)
        at me.coley.recaf.Recaf.loadPlugins(Recaf.java:108)
        at me.coley.recaf.Recaf.launch(Recaf.java:93)
        at me.coley.recaf.Recaf.main(Recaf.java:47)
Caused by: java.lang.NoSuchFieldException: parent
        at java.base/java.lang.Class.getDeclaredField(Class.java:2417)
        at me.coley.recaf.util.VMUtil.setParent(VMUtil.java:149)
        ... 6 common frames omitted
Exception in thread "WindowsNativeRunloopThread" java.lang.NoSuchMethodError: <init>
        at com.sun.glass.ui.win.WinApplication.staticScreen_getScreens(Native Method)
        at com.sun.glass.ui.Screen.initScreens(Screen.java:412)
        at com.sun.glass.ui.Application.lambda$run$1(Application.java:152)
        at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
        at com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(WinApplication.java:174)
        at java.base/java.lang.Thread.run(Thread.java:835)
Exception in thread "JavaFX Application Thread" java.lang.NullPointerException
        at com.sun.prism.d3d.D3DPipeline.getAdapterOrdinal(D3DPipeline.java:205)
        at com.sun.javafx.tk.quantum.QuantumToolkit.assignScreensAdapters(QuantumToolkit.java:738)
        at com.sun.javafx.tk.quantum.QuantumToolkit.runToolkit(QuantumToolkit.java:334)
        at com.sun.javafx.tk.quantum.QuantumToolkit.lambda$startup$10(QuantumToolkit.java:279)
        at com.sun.glass.ui.Application.lambda$run$1(Application.java:153)
        at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
        at com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(WinApplication.java:174)
        at java.base/java.lang.Thread.run(Thread.java:835)
```

--- a/src/main/java/me/coley/recaf/Recaf.java
+++ b/src/main/java/me/coley/recaf/Recaf.java
@@ -8,6 +8,7 @@ import me.coley.recaf.plugin.PluginsManager;
 import me.coley.recaf.plugin.api.EntryLoaderProviderPlugin;
 import me.coley.recaf.util.Log;
 import me.coley.recaf.util.Natives;
+import me.coley.recaf.util.VMUtil;
 import me.coley.recaf.util.self.SelfDependencyPatcher;
 import me.coley.recaf.util.self.SelfUpdater;
 import me.coley.recaf.workspace.InstrumentationResource;
@@ -72,6 +73,8 @@ public class Recaf {
 	 */
 	private static void init() {
 		if (!initialized) {
+			// Bypass JDK restrictions.
+			VMUtil.patch();
 			// Patch in dependencies
 			SelfDependencyPatcher.patch();
 			// Fix title bar not displaying in GTK systems

--- a/src/main/java/me/coley/recaf/util/ClasspathUtil.java
+++ b/src/main/java/me/coley/recaf/util/ClasspathUtil.java
@@ -186,7 +186,7 @@ public class ClasspathUtil {
 	}
 
 	private static Set<String> scanBootstrapClasses() throws Exception {
-		float vmVersion = VMUtil.getVmVersion();
+		int vmVersion = VMUtil.getVmVersion();
 		Set<String> classes = new LinkedHashSet<>(4096, 1F);
 		if (vmVersion < 9) {
 			Method method = ClassLoader.class.getDeclaredMethod("getBootstrapClassPath");

--- a/src/main/java/me/coley/recaf/util/VMUtil.java
+++ b/src/main/java/me/coley/recaf/util/VMUtil.java
@@ -1,5 +1,6 @@
 package me.coley.recaf.util;
 
+import com.nqzero.permit.Permit;
 import com.sun.javafx.application.PlatformImpl;
 import javafx.application.Platform;
 
@@ -11,6 +12,7 @@ import java.net.URL;
 import java.net.URLClassLoader;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.util.HashMap;
 
 /**
  * Dependent and non-dependent platform utilities for VM.
@@ -201,7 +203,34 @@ public final class VMUtil {
             case MAC:
                 return bin.resolve("java");
             default:
-                throw new IllegalArgumentException("Don't know how  to find Java path for: " + os.getMvnName());
+                throw new IllegalArgumentException("Don't know how to find Java path for: " + os.getMvnName());
+        }
+    }
+
+    /**
+     * Patches JDK stuff.
+     */
+    public static void patch() {
+        if (getVmVersion() > 8) {
+            Permit.godMode();
+            Permit.unLog();
+            Class<?> klass;
+            try {
+                klass = Class.forName("jdk.internal.reflect.Reflection",
+                        true, null);
+            } catch (ClassNotFoundException ex) {
+                throw new IllegalStateException("Unable to locate 'jdk.internal.reflect.Reflection' class", ex);
+            }
+            try {
+                Field field = klass.getDeclaredField("fieldFilterMap");
+                field.setAccessible(true);
+                field.set(null, new HashMap<>(0));
+                field = klass.getDeclaredField("methodFilterMap");
+                field.setAccessible(true);
+                field.set(null, new HashMap<>(0));
+            } catch (IllegalAccessException | NoSuchFieldException ex) {
+                throw new IllegalStateException("Unable to patch reflection filters", ex);
+            }
         }
     }
 }

--- a/src/main/java/me/coley/recaf/util/VMUtil.java
+++ b/src/main/java/me/coley/recaf/util/VMUtil.java
@@ -150,7 +150,7 @@ public final class VMUtil {
         try {
             field = ClassLoader.class.getDeclaredField("parent");
         } catch (NoSuchFieldException ex) {
-            throw new IllegalStateException("No 'parent' field in java.lang.ClassLoader", ex);
+            throw new RuntimeException("No 'parent' field in java.lang.ClassLoader", ex);
         }
         field.setAccessible(true);
         try {
@@ -171,7 +171,7 @@ public final class VMUtil {
             try {
                 m = Platform.class.getDeclaredMethod("startup", Runnable.class);
             } catch (NoSuchMethodException ex) {
-                throw new IllegalStateException("javafx.application.Platform.startup(Runnable) is missing", ex);
+                throw new RuntimeException("javafx.application.Platform.startup(Runnable) is missing", ex);
             }
             m.setAccessible(true);
             try {
@@ -179,7 +179,7 @@ public final class VMUtil {
             } catch (IllegalAccessException ex) {
                 throw new IllegalStateException("'startup' became inaccessible", ex);
             } catch (InvocationTargetException ex) {
-                throw new IllegalStateException("Unable to initialize toolkit", ex.getTargetException());
+                throw new RuntimeException("Unable to initialize toolkit", ex.getTargetException());
             }
         }
     }
@@ -219,7 +219,7 @@ public final class VMUtil {
                 klass = Class.forName("jdk.internal.reflect.Reflection",
                         true, null);
             } catch (ClassNotFoundException ex) {
-                throw new IllegalStateException("Unable to locate 'jdk.internal.reflect.Reflection' class", ex);
+                throw new RuntimeException("Unable to locate 'jdk.internal.reflect.Reflection' class", ex);
             }
             try {
                 Field field = klass.getDeclaredField("fieldFilterMap");
@@ -229,7 +229,7 @@ public final class VMUtil {
                 field.setAccessible(true);
                 field.set(null, new HashMap<>(0));
             } catch (IllegalAccessException | NoSuchFieldException ex) {
-                throw new IllegalStateException("Unable to patch reflection filters", ex);
+                throw new RuntimeException("Unable to patch reflection filters", ex);
             }
         }
     }

--- a/src/main/java/me/coley/recaf/util/self/SelfDependencyPatcher.java
+++ b/src/main/java/me/coley/recaf/util/self/SelfDependencyPatcher.java
@@ -1,6 +1,5 @@
 package me.coley.recaf.util.self;
 
-import com.nqzero.permit.Permit;
 import me.coley.recaf.Recaf;
 import me.coley.recaf.util.ClasspathUtil;
 import me.coley.recaf.util.Log;
@@ -123,8 +122,6 @@ public class SelfDependencyPatcher {
 				Log.error(ex, "Failed to convert '%s' to URL", path.toFile().getAbsolutePath());
 			}
 		});
-		// Bypass the access system
-		Permit.godMode();
 		// Fetch UCP of application's ClassLoader
 		// - ((ClassLoaders.AppClassLoader) ClassLoaders.appClassLoader()).ucp
 		Class<?> clsClassLoaders = Class.forName("jdk.internal.loader.ClassLoaders");

